ccpage178.py

```diff
@@ -0,0 +1,179 @@
+    def __init__(self, first_name, last_name):
+        """Initialize the attributes from parent to child"""
+        super().__init__(first_name, last_name)
+        self.priviledges = ['can add post', 'can delete post', 'can ban user']
+    
+    def show_admin_priviledges(self):
+        """Show the list of admin priviledges"""
+        print("These are the privledges of an Admin: ")
+        for priviledge in self.priviledges:
+            print(f"-{priviledge.capitalize()}")
+
+
+user_type = Admin('tom', 'smith')
+
+print(user_type.show_admin_priviledges())
+
+
+
+# Based on the code above:
+# Write a separate Privileges class. 
+# The class should have one attribute, privileges, that stores a list of strings as described in exercise above. 
+# Move the show_privileges() method to this class.
+# Make a Privileges instance as an attribute in the Admin class.
+# Create a new instance of Admin and use your method to show its privileges.
+class User():
+    """"An attempt to model a user"""
+    def __init__(self, first_name, last_name):
+        """Initialize the attributes"""
+        self.first_name = first_name
+        self.last_name = last_name
+    
+    def describe_user(self):
+        """Summarize the user's first and last names"""
+        full_name = self.first_name.title() + ' ' + self.last_name.title()
+        return full_name
+    
+class Priviledges():
+    """An attempt to list the priviledges for a user"""
+    def __init__(self, priviledges=['create expense reports', 'organize company meetings', 'plan company events']):
+        self.priviledges = priviledges
+    
+    def show_priviledges(self):
+        """Show the priviledges"""
+        print("These are the priviledges available: ")
+        for priviledge in self.priviledges:
+            print(f"-{priviledge.capitalize()}")
+
+class Admin(User):
+    """An attempt to model a specific type of user, an admin"""
+    def __init__(self, first_name, last_name):
+        """Initialize the attributes from parent to child"""
+        super().__init__(first_name, last_name)
+        self.priviledges_available = Priviledges()
+    
+
+user_type = Admin('tom', 'smith')
+
+print(user_type.priviledges_available.show_priviledges())
+
+
+
+# Add a method to the Battery class called upgrade_battery().
+# This method should check the battery size and set the capacity to 85 if it isn’t already.
+# Make an electric car with a default battery size, call get_range() once, and then call get_range() a second time after upgrading the battery.
+class Car():
+    """An attempt to model a car"""
+    def __init__(self, make, model, year):
+        self.make = make
+        self.model = model
+        self.year = year
+        self.odometer_reading = 100
+
+class Battery():
+    """An attempt to provide detail about a battery for a specific car, an electric one"""
+    def __init__(self, battery=70):
+        """Initialize the attribute"""
+        self.battery = battery
```

Comment:
I'd add an instance variable for the `range` that will store it just like the `battery_size`.

Reply:
`amy`

```
lkasdjf
lasdflafsd
```